File: 백준/Bronze/9046. 복호화/복호화.js
const fs = require('fs');
// let input = fs.readFileSync('./input.txt').toString().trim().split('\n');

/**정답 제출 시에는 이 코드로 제출 */
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

let n = input.shift();
for(let i=0; i<n; i++){
    let target = input[i].trim().split(" ")
    let t_map = new Map();
    let a = []

    target.forEach((word) => {
        for(let j=0; j<word.length; j++){
            t_map.set(word[j],(t_map.get(word[j])||0)+1);
        }
    })
    
    let max_val = Math.max(...t_map.values());
    t_map.forEach((value,key) => {
        if(value === max_val) {
            a.push(key);
        }
    });
    
    if(([...t_map.values()].length) === 0 || a.length>1){
        console.log('?');
    } else {
        console.log(a[0]);
    }
}

File: 백준/Bronze/9046. 복호화/복호화.py
import sys
input = sys.stdin.readline

n = int(input().rstrip())
for _ in range(n):
    answer = []
    t_dict = {}

    target = input().rstrip()
    t_arr = target.split(' ')

    for t in t_arr:
        for i in t:
            if i not in t_dict:
                t_dict[i] = 1
            else:
                t_dict[i] += 1
    
    answer = sorted(t_dict.values() ,reverse=True)
    if (len(answer)>1 and answer[0]==answer[1]) or len(answer) == 0:
        print('?')
    else:
        print(max(t_dict,key=t_dict.get))